src/erf.jl

```diff
@@ -362,6 +356,43 @@ function _erfinv(x::Float32)
     end
 end
 
+function _erfinv(x::Float16)
+    a = abs(x)
+    if a > Float16(1)
+        throw(DomainError(a, "`abs(x)` cannot be greater than 1."))
+    elseif a == Float16(1)
+        return copysign(Inf16, x)
+    else
+        # Perform calculations with `Float32`
+        x32 = Float32(x)
+        a32 = Float32(a)
+        if a32 <= 0.75f0 # Table 7 in Blair et al.
+            t = x32^2 - 0.5625f0
+            y = x32 * @horner(t, -0.10976_672f1,
+                              0.53062_1f0) /
+                      @horner(t, -0.10123_953f1,
+                              0.1f1)
```

Comment:
```suggestion
t = muladd(-6.73815f1, x32, 1f0) / muladd(-4.18798f0, x32, 4.54263f0)
y = copysign(muladd(0.886226925453, x, t), x32)
```
I believe this is strictly higher accuracy for the same number of arithmetic operations and 1 fewer constant.

Reply:
(yes I did get this by playing around on desmos for ~30 minutes) https://www.desmos.com/calculator/f3gwnufktv

Reply:
Did you mean to write
```suggestion
        if a32 <= 0.75f0
            # Simpler and more accurate alternative to Table 7 in Blair et al.
            # Ref: https://github.com/JuliaMath/SpecialFunctions.jl/pull/372#discussion_r1592832735
            t = muladd(-6.73815f1, x32, 1f0) / muladd(-4.18798f0, x32, 4.54263f0)
            y = copysign(muladd(0.88622695f0, x32, t), x32)
```
or was it intentional that you used `x` instead of `x32` in the definition of `y`?

Reply:
Your version is the right one.